cannot build documentation and hence build fails

`make docs` results in
 ```julia
Building HTML documentation.
ERROR: LoadError: ArgumentError: Package Pkg not found in current path:
 - Run `Pkg.add("Pkg")` to install the Pkg package.
Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:816
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1034
 [4] include(::Module, ::String) at ./sysimg.jl:29
 [5] exec_options(::Base.JLOptions) at ./client.jl:234
 [6] _start() at ./client.jl:427
in expression starting at /home/jagan/srces/julia-git/src/julia/doc/make.jl:2
make[2]: *** [Makefile:38: html] Error 1
make[2]: Leaving directory '/home/jagan/srces/julia-git/src/julia/doc'
make[1]: *** [Makefile:88: docs] Error 2
make[1]: Leaving directory '/home/jagan/srces/julia-git/src/julia'
Make: *** [Makefile:50: /home/jagan/srces/julia-git/src/julia/doc/_build/html/en/index.html] Error 2 
```
This is also run as part of `make install`. `make` finished with a the julia binary in `/home/jagan/srces/julia-git/src/julia/usr/bin/julia` so only documentation fails to build.

It fails at a different place now, 
```julia
Building HTML documentation.
LOAD_PATH = String[]
  Updating registry at `deps/registries/Uncurated`
  Updating git-repo `https://github.com/JuliaRegistries/Uncurated.git`
ERROR: ArgumentError: Package OldPkg not found in current path:
 - Run `Pkg.add("OldPkg")` to install the OldPkg package.

Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:816
 [2] top-level scope at none:2
 [3] eval at ./boot.jl:319 [inlined]
 [4] eval(::Expr) at ./client.jl:394
 [5] top-level scope at ./none:3 [inlined]
 [6] top-level scope at ./<missing>:0
ERROR: LoadError: Failed to precompile Documenter to /home/jagan/srces/julia-git/src/julia/doc/deps/compiled/v0.7/Documenter/sHGj.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:298 [inlined]
 [3] compilecache(::Base.PkgId) at ./loading.jl:1169
 [4] _require(::Base.PkgId) at ./loading.jl:942
 [5] require(::Base.PkgId) at ./loading.jl:838
 [6] require(::Module, ::Symbol) at ./loading.jl:833
 [7] include at ./boot.jl:317 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1034
 [9] include(::Module, ::String) at ./sysimg.jl:29
 [10] exec_options(::Base.JLOptions) at ./client.jl:234
 [11] _start() at ./client.jl:427
in expression starting at /home/jagan/srces/julia-git/src/julia/doc/make.jl:10
make[1]: *** [Makefile:38: html] Error 1
make: *** [Makefile:88: docs] Error 2
```

--- a/doc/Project.toml
+++ b/doc/Project.toml
@@ -1,3 +1,2 @@
 [deps]
-Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"

--- a/doc/make.jl
+++ b/doc/make.jl
@@ -1,8 +1,9 @@
 # Install dependencies needed to build the documentation.
+empty!(LOAD_PATH)
+push!(LOAD_PATH, @__DIR__, "@stdlib")
 using Pkg
 empty!(DEPOT_PATH)
 pushfirst!(DEPOT_PATH, joinpath(@__DIR__, "deps"))
-pushfirst!(LOAD_PATH, @__DIR__)
 Pkg.instantiate()
 
 using Documenter

